app/src/main/java/org/fossasia/openevent/general/event/EventsDataSourceFactory.kt

```diff
@@ -0,0 +1,20 @@
+package org.fossasia.openevent.general.event
+
+import androidx.lifecycle.MutableLiveData
+import androidx.paging.DataSource
+import io.reactivex.disposables.CompositeDisposable
+import org.fossasia.openevent.general.event.faq.EventsDataSource
+
+class EventsDataSourceFactory(private val compositeDisposable: CompositeDisposable,
+                              private val eventService: EventService,
+                              private val query:String?)
```

Comment:
You are already doing it here

Reply:
I thought it can be some better way of doing this, i was not satisfied with the current implementation

Reply:
Factory means you can take any argument to produce other type